### PR TITLE
Add /proc/io/num-open-file-descriptors built-in metric.

### PR DESCRIPTION
**Pull request by rgooch**

### CODE REVIEW

File: go/tricorder/setup.go [hunk 1]
```diff
@@ -116,6 +118,14 @@ func initDefaultMetrics() {
 		resourceUsageGroup,
 		units.None,
 		"Block input operations")
+	if numOpenFileDescriptors >= 0 {
+		RegisterMetricInGroup(
+			"/proc/io/num-open-file-descriptors",
+			&numOpenFileDescriptors,
```
Comment (keep94) [inline go/tricorder/setup.go]: Alternatively, you could just pass countOpenFileDescriptors function pointer and keep this metric out of the group. 

Keep in mind that each time you query for this metric, it causes the above RegisterUpdateFunc for the group to be called. Using groups is good if you must use a system call that returns several usage metrics at once. In this case, you don't have that.

**Review by keep94 — COMMENTED** [review]
I reviewed the code.

**Comment by rgooch** [comment]
PTAL.

File: go/tricorder/setup.go [hunk 2]
```diff
@@ -118,10 +116,10 @@ func initDefaultMetrics() {
 		resourceUsageGroup,
 		units.None,
 		"Block input operations")
-	if numOpenFileDescriptors >= 0 {
+	if countOpenFileDescriptors() >= 0 {
 		RegisterMetricInGroup(
 			"/proc/io/num-open-file-descriptors",
-			&numOpenFileDescriptors,
```
Comment (keep94) [inline go/tricorder/setup.go]: Consider just using RegisterMetric rather than RegisterMetricInGroup.

**Comment by rgooch** [comment]
PTAL #2.

**Review by keep94 — APPROVED** [review]

**Pull request merged**